ote/src/clj/ote/nap/publish.clj

```diff
@@ -0,0 +1,111 @@
+(ns ote.nap.publish
+  "Publish transport-service as CKAN dataset."
+  (:require [ote.nap.ckan :as ckan]
+            [ote.db.transport-operator :as t-operator]
+            [ote.db.transport-service :as t-service]
+            [ote.db.operation-area :as operation-area]
+            [specql.core :refer [fetch] :as specql]
+            [clojure.string :as str]
+            [taoensso.timbre :as log]))
+
+(defn- fetch-service-operation-area-description
+  "Fetch the operation area as a comma separated list (for SOLR facet search).
+  Takes all operation areas for the transport service and joins their description
+  texts into a single string."
+  [db service-id]
+  (->> (fetch db ::operation-area/operation-area
+              #{::operation-area/description}
+              {::operation-area/transport-service-id service-id})
+       (mapcat ::operation-area/description)
+       (map ::t-service/text)
+       (str/join ", ")))
+
+(defn- ckan-dataset-description
+  "Create a CKAN dataset description that can be used with the CKAN API to
+  create a dataset. The owner organization is the user's organization."
+  [db user {service-id ::t-service/id :as ts}]
+  {:ckan/name (str "org-" (::t-service/transport-operator-id ts)
+                   "-service-" service-id)
+   :ckan/title (::t-service/name ts)
+   :ckan/owner-org (get-in user [:group :name])
+   :ckan/transport-service-type (name (::t-service/type ts))
+   :ckan/operation-area (fetch-service-operation-area-description db service-id)})
+
+(defmulti interface-description ::t-service/type)
+
+(defmethod interface-description :passenger-transportation [ts]
+  {:ckan/url (str "/ote/export/geojson/"
+                  (::t-service/transport-operator-id ts) "/"
+                  (::t-service/id ts))
+   :ckan/name (str (::t-service/name ts) " GeoJSON")
+   :ckan/format "GeoJSON"})
+
+(defn- ckan-resource-description
+  "Create a CKAN resource description that can be used with the CKAN API to
+  create a resource."
+  [export-base-url ts {:ckan/keys [id name] :as ckan-dataset}]
+  (println "CKAN-DATASET: " (pr-str ckan-dataset))
```

Comment:
What is the difference between println and (log/error ?

Reply:
I often use println for stuff I definitely intend to delete before merging